shopfloor/services/service.py

```diff
@@ -46,18 +46,19 @@ def search_move_line(self):
         # every time.
         return self._actions_for("search_move_line", picking_types=self.picking_types)
 
-    def _check_picking_status(self, pickings):
+    def _check_picking_status(self, pickings, states=None):
         """Check if given pickings can be processed.
 
         If the picking is already done, canceled or didn't belong to the
         expected picking type, a message is returned.
         """
+        valid_states = states or ["assigned"]
```

Comment:
could be done in the function declaration ?
`def _check_picking_status(self, pickings, states=["assigned"]):`

Reply:
this is bad practice to use mutable objects as default values:
```python
❯ python3
Python 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def func(arg=[]):
...     arg.append("potato")
...     print(arg)
... 
>>> func()
['potato']
>>> func()
['potato', 'potato']
>>> func()
['potato', 'potato', 'potato']
```

Reply:
yep: never use non hashable types as default args

Reply:
Then why not using a tuple here? Auto-documented, no need to check the implementation.
```python
def _check_picking_status(self, pickings, states=("assigned",)):
```

Reply:
My comment still apply here.